### ISSUE TITLE
Veinminer Issues

### ISSUE THREAD

**Issue by Rhythm-Boss** [scalda/FoolCraft#234]
Hi, I recently tried starting a server with a friend of mine for us to play on. I downloaded game version 1.2.2 from Curse, and server version 1.2.1 (while copying the client's config folder over as it said to do). Everything worked fine at first, except that Veinminer, despite editing config, both server-side and client-side, would not work with the bound key OR with sneak-mode enabled. Simply nothing would happen. Then, after not playing on the server for a day, and trying again to change the server and client config file, the game would crash on loading mods part 1 when it hit vein miner, until i restored the config file to the original. Then after I joined the game, it would still not work until I tried sneak-mode, which then caused the server to crash. I can and will post the crash logs for the server. I appreciate any help, and really wish to get veinminer working. Thanks in advance. 
Also, apologies if this was poorly explained, I'm very tired! :D

**Comment by Rhythm-Boss** [comment]
server crash: 
---- Minecraft Crash Report ----
// I let you down. Sorry :(

Time: 4/9/17 5:35 PM
Description: Exception ticking world

java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.remove(Unknown Source)
	at lr.d(SourceFile:198)
	at ls.d(SourceFile:202)
	at net.minecraft.server.MinecraftServer.D(SourceFile:618)
	at ld.D(SourceFile:338)
	at net.minecraft.server.MinecraftServer.C(SourceFile:554)
	at net.minecraft.server.MinecraftServer.run(SourceFile:458)
	at java.lang.Thread.run(Unknown Source)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at java.util.HashMap$HashIterator.remove(Unknown Source)
	at lr.d(SourceFile:198)
	at ls.d(SourceFile:202)

-- Affected level --
Details:
	Level name: FoolCraft
	All players: 0 total; []
	Chunk stats: ServerChunkCache: 791 Drop: 536
	Level seed: 8615504990733332294
	Level generator: ID 00 - default, ver 1. Features enabled: true
	Level generator options: 
	Level spawn location: World: (112,64,252), Chunk: (at 0,4,12 in 7,15; contains blocks 112,0,240 to 127,255,255), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 146775 game time, 38194 day time
	Level dimension: 0
	Level storage version: 0x04ABD - Anvil
	Level weather: Rain time: 8466 (now: true), thunder time: 3141 (now: true)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
Stacktrace:
	at net.minecraft.server.MinecraftServer.D(SourceFile:618)
	at ld.D(SourceFile:338)
	at net.minecraft.server.MinecraftServer.C(SourceFile:554)
	at net.minecraft.server.MinecraftServer.run(SourceFile:458)
	at java.lang.Thread.run(Unknown Source)

-- System Details --
Details:
	Minecraft Version: 1.10.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_111, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 562090320 bytes (536 MB) / 836763648 bytes (798 MB) up to 2844786688 bytes (2713 MB)
	JVM Flags: 0 total; 
	IntCache: cache: 0, tcache: 0, allocated: 13, tallocated: 95
	Profiler Position: N/A (disabled)
	Player Count: 0 / 20; []
	Is Modded: Unknown (can't tell)
	Type: Dedicated Server (map_server.txt)

**Comment by CakeArmy** [comment]
Im not understanding how veinminer doesn't work client side? are you still having this issue?

**Comment by Rhythm-Boss** [comment]
My apologies for the delay, I've been quite busy with other things. As for the issue, I haven't experienced anymore crashes since completely reinstalling all server and client files, but still veinminer refuses to work. I try using "/veinminerc pressed" on my client to use the bound key [grave] to activate the effect, and I've also tried "/veinminerc sneak" which also does not work. I have not touched any client config files, and all i've done is copied the client config folder onto the server files (as per instructions for the 1.2.2 update). On a side note, I'm experience "Warn: Server can't keep up, running # ms behind, skipping # ticks", despite having setup a .bat file to run the server with 4GB of RAM (unless I've set this up wrong, which I doubt because the console window shows the server using a few hundred mB, while having 98% memory free). I'll gladly provide any more info you need per your request.

**Comment by CakeArmy** [comment]
Oh. I know the problem. See this thread I made.

**Comment by CakeArmy** [comment]
Check out this thread @Rhythm-Boss 
https://github.com/scalda/FoolCraft/issues/286

**Comment by Rhythm-Boss** [comment]
You don't understand.There is literally nothing in the world. After complete fresh install, joining the brand new world and mining a single block (have tested with multiple seeds) the same thing happens. This lag can't be from "something in the world taking too long to tick" because nothing exists that would do such a thing.

**Comment by CakeArmy** [comment]
Ohh

**Comment by CakeArmy** [comment]
Hmm... Do you want to send me your server files? Do you want me to send you server files that work?

**Comment by Rhythm-Boss** [comment]
I mean sure, I guess it couldn't hurt to try. I assume you haven't touched the configs?

**Comment by CakeArmy** [comment]
I would give you a fresh server correct.

**Comment by CakeArmy** [comment]
Ill redownload, update config for ver 1.2.2 make sure it runs and send over. How much ram do you want it to have? start.bat okay? can you convert to .whatever linux is cant remember

**Comment by Rhythm-Boss** [comment]
start.bat is very much ok, I was thinking about 4 GB RAM for the server, but I can always edit the file. Thanks in advance, my buddy actually wants to play tonight, so if this gets working, then wonderful.

**Comment by CakeArmy** [comment]
ill do it right now, one minute

**Comment by CakeArmy** [comment]
Its uploading now.

**Comment by Rhythm-Boss** [comment]
cool, can't wait

**Comment by CakeArmy** [comment]
Sorry. Not sure if this is going to finish tonight. I have fast upload speed but its a bunch of files and dropbox limits how fast I can upload anyways.

**Comment by Rhythm-Boss** [comment]
did you not compress the folder?

**Comment by CakeArmy** [comment]
I did
It took awhile but here ya go. Sorry for the delay. https://www.dropbox.com/sh/oi2flnhfxdixqbc/AAAk2bMt-KbT9qEPn-FV5Qqva?dl=1

**Comment by CakeArmy** [comment]
*facepalm* I uploaded the uncompressed version. You want it compressed?

**Comment by Rhythm-Boss** [comment]
no problem, thanks for the effort. no worries, i was just thinking for your upload's sake, i have amazing download. I'll get back to you when i finish trying it and let you know

**Comment by Rhythm-Boss** [comment]
So it turns out you did in fact upload the compressed folder, so no worries about that. I have unzipped and ran all the files, and I can confirm that everything works. However, I find that the server is using about 70% of CPU, and my client is using more memory than I thought I allocated to it, and there's much lag on the server. However, veinminer and all other mods do in fact work, and there is no "Server can't keep up", just block lag, so I can't tell the cause in this case. Also, you put "10024M" for the run.bat file lol, so I adjusted it to 4096M for 4GB :P

**Comment by scalda** [comment]
Hi,

1.3 has just been released. Can you please retest this problem and if it still persists can you open a new issue using the template provided.

Regards

Scalda

**Comment by Rhythm-Boss** [comment]
@scalda, I see no download anywhere (on github or the curse launcher) for the 1.3 files.

**Comment by Whazaaaaa** [comment]
I'd like to add that some people trying to join get mod rejection for veinminer. And when I had veinminer enabled (some people somehow were able to join even with the rejection) my server would time-out when someone veinmined 'something'. So I had to remove veinminer from my server and all my problems were gone.
On a side note:
```25.04 10:44:24 [Server] Netty Epoll Server IO #4/ERROR [FML]: NetworkDispatcher exception
25.04 10:44:24 [Server] INFO io.netty.handler.timeout.ReadTimeoutException```

This is what I got that timed-out my server when veinminer was still on the server. To me its useless and probably to you aswell but who knows maybe it gives some insight.
